nsot/middleware/auth.py

```diff
@@ -50,3 +51,31 @@ def configure_user(self, user):
 
         log.debug('Created new user: %s', user)
         return user
+
+
+class NsotObjectPermissionsBackend(ObjectPermissionBackend):
+    """Custom backend that overloads django-guardian's has_perm method."""
+    def has_perm(self, user_obj, perm, obj=None):
+        """
+        Returns ``True`` if ``grp_obj`` has ``perm`` for ``obj``. If no
+        ``obj`` is provided, ``False`` is returned.
+
+        However, if ``grp_obj`` does not have ``perm`` for ``obj``,
+        the ancestor tree for ``obj`` is checked against. If any node in
+        the ancestor tree has ``perm`` for the ``obj``, then ``True`` is
+        returned, else ``False`` is returned.
+        """
+        check = super(NsotObjectPermissionsBackend,self).has_perm(
+            user_obj, perm, obj
+        )
+        if check:
+            return True
+
+        if hasattr(obj, 'get_ancestors', False):
+            ancestors = obj.get_ancestors()
+
+            return any(check.has_perm(perm, a) for a in ancestors)
+
+        # (TODO nseshan): Need to modify the logic for obj without ancestors
```

Comment:
I don't think we need to modify anything. We could just switch this to a bare `return False`.

Reply:
Done.

Reply:
If the object doesn't have ancestors, then the initial permission check pass/fail is sufficient. Which means if that initial check returns `True` nothing left to do, and if it's `False` and doesn't have `.get_ancestors` it's still `False`. ~fin~

Reply:
Fair point! 👍